mmrazor/models/algorithms/space_mixin.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Dict
+
+import torch
+import torch.nn as nn
+
+from mmrazor.models.mutables import BaseMutable
+from mmrazor.models.mutators import DiffModuleMutator
+
+
+class SpaceMixin():
+    """A mixin for :class:`BaseAlgorithm`, which can unify ``search_groups``
+    from different mutators into the overall search_space with providing APIs
+    for subnet operations.
+
+    Note:
+        SpaceMixin is designed for handling different type of mutators,
+        therefore it currently supports NAS/Pruning algorithms with mutator(s).
+    """
+
+    def _build_search_space(self, prefix=''):
```

Comment:
Add necessary hint for users that `mutator.prepare_from_supernet` has to be called before `_build_search_space`.